### PR TITLE
fix(2911): add validation in federation resolvers

### LINKED ISSUE #2911: Federation Subgraph: add validation for used fields
Resolvers on types could use different fields from the respective type inside the resolver itself. 

Add a validation that every field that is used as `.value.field_name` inside the resolver is actually part of the type and if not generated a Validation error with info about unknown field. Make sure nested paths `.value.a.b.c` are covered as well.

**Good Example**

```graphql
type User @http(path: "/users/{{.value.id}}") {
  id: ID
  name: String
}
```

**Bad Examples**

```graphql
type User @http(path: "/users/{{.value.foo.id}}") {
  id: ID
  name: String
}
```

```graphql
type User @http(path: "/users/{{.value.age}}") {
  id: ID
  name: String
}
```

### PR DESCRIPTION
**Pull request by dekkku**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
/claim #2911
Fixes #2911

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/config/transformer/subgraph.rs [hunk 1]
```diff
@@ -192,6 +194,68 @@ fn combine_keys(v: Vec<Keys>) -> Keys {
...
+            if let Some(type_def) = config.types.get(current_type) {
+                if !type_def.fields.contains_key(key) {
+                    return Valid::fail(format!(
+                        "Invalid key at index {}: '{}' is not a field of '{}'",
+                        index, key, current_type
+                    ));
+                }
+                current_type = type_def.fields[key].type_of.name();
+            } else {
+                return Valid::fail(format!("Type '{}' not found in config", current_type));
```
Comment (tusharmath) [inline src/core/config/transformer/subgraph.rs:232]: Need a test for this new case.

**Comment by tusharmath** [comment]
@dekkku Apologies for the delay. I think we just need one more test. Otherwise looks good to me. Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

File: src/core/config/transformer/subgraph.rs [hunk 2]
```diff
@@ -39,22 +39,24 @@ impl Transform for Subgraph {
             // if federation is disabled don't process the config
             return Valid::succeed(config);
         }
-
+        let new_config = config.clone();
```
Comment (laststylebender14) [inline src/core/config/transformer/subgraph.rs:42]: if we just want to check the availability of types, then don't clone the entire config just clone the types map.

File: src/core/config/transformer/subgraph.rs [hunk 3]
```diff
@@ -192,6 +194,68 @@ fn combine_keys(v: Vec<Keys>) -> Keys {
...
+                if expr.len() > 1 && expr[0].as_str() == "value" {
+                    Self::validate_iter(config, type_name, expr.iter().skip(1))
+                } else {
+                    Valid::succeed(())
+                }
+            } else {
+                Valid::succeed(())
+            }
+        })
+        .map(|_| ())
```
Comment (laststylebender14) [inline src/core/config/transformer/subgraph.rs:213]:
```suggestion
        .unit()
```

File: src/core/config/transformer/subgraph.rs [hunk 4]
```diff
@@ -192,6 +194,68 @@ fn combine_keys(v: Vec<Keys>) -> Keys {
...
+                        index, key, current_type
+                    ));
+                }
+                current_type = type_def.fields[key].type_of.name();
+            } else {
+                return Valid::fail(format!("Type '{}' not found in config", current_type));
+            }
+            Valid::succeed(())
+        })
+        .map(|_| ())
```
Comment (laststylebender14) [inline src/core/config/transformer/subgraph.rs:236]:
```suggestion
        .unit()
```

File: src/core/config/transformer/subgraph.rs [hunk 5]
```diff
@@ -192,6 +194,68 @@ fn combine_keys(v: Vec<Keys>) -> Keys {
...
+                    config,
+                    Mustache::parse(&q.value).segments().iter(),
+                )
+            })
+            .and(Self::validate_expressions(
+                type_name,
+                config,
+                Mustache::parse(&http.url).segments().iter(),
+            ))
+            .map(|_| ())
```
Comment (laststylebender14) [inline src/core/config/transformer/subgraph.rs:253]:
```suggestion
            .unit()
```

**Review by tusharmath — APPROVED** [review]

**Pull request merged**